server/src/main/java/org/candlepin/dto/api/v1/ProductTranslator.java

```diff
@@ -61,54 +64,94 @@ public ProductDTO populate(Product source, ProductDTO destination) {
-        destination.setName(source.getName());
-        destination.setMultiplier(source.getMultiplier());
-        destination.setHref(source.getHref());
-        destination.setLocked(source.isLocked());
-        destination.setAttributes(source.getAttributes());
-        destination.setDependentProductIds(source.getDependentProductIds());
+        destination.created(Util.toDateTime(source.getCreated()))
+            .updated(Util.toDateTime(source.getUpdated()))
+            .uuid(source.getUuid())
+            .id(source.getId())
+            .name(source.getName())
+            .multiplier(source.getMultiplier())
+            .href(source.getHref())
+            .locked(source.isLocked())
+            .attributes(toAttributes(source.getAttributes()))
+            .productContent(new HashSet<>())
+            .branding(new HashSet<>())
+            .dependentProductIds(new HashSet<>(source.getDependentProductIds()));
 
         if (modelTranslator != null) {
             Collection<ProductContent> productContent = source.getProductContent();
-            destination.setProductContent(Collections.emptyList());
-
             if (productContent != null) {
-                ObjectTranslator<Content, ContentDTO> contentTranslator = modelTranslator
-                    .findTranslatorByClass(Content.class, ContentDTO.class);
+                ObjectTranslator<ProductContent, ProductContentDTO> contentTranslator = modelTranslator
+                    .findTranslatorByClass(ProductContent.class, ProductContentDTO.class);
 
+                Set<ProductContentDTO> content = new HashSet<>();
                 for (ProductContent pc : productContent) {
                     if (pc != null) {
-                        ContentDTO dto = contentTranslator.translate(modelTranslator, pc.getContent());
-
+                        ProductContentDTO dto = contentTranslator.translate(modelTranslator, pc);
                         if (dto != null) {
-                            destination.addContent(dto, pc.isEnabled());
+                            content.add(dto);
                         }
                     }
                 }
+                destination.productContent(content);
+            }
+            else {
+                destination.productContent(Collections.emptySet());
             }
 
             Collection<Branding> branding = source.getBranding();
             if (branding != null && !branding.isEmpty()) {
+                Set<BrandingDTO> dtos = new HashSet<>();
                 for (Branding brand : branding) {
                     if (brand != null) {
-                        destination.addBranding(modelTranslator.translate(brand, BrandingDTO.class));
+                        BrandingDTO dto = modelTranslator.translate(brand, BrandingDTO.class);
+                        if (isNullOrIncomplete(dto)) {
+                            throw new IllegalArgumentException("branding is null or incomplete");
```

Comment:
I don't think we should be throwing exceptions in translators. This kind of thing is for when we receive a DTO in a POST/PUT request and we have to validate the client data. Here, we're just translating the data we know is already valid (Branding) to be output back, so we don't need any validation

Reply:
Done.